client/sites-dashboard/components/sites-dashboard.tsx

```diff
@@ -134,21 +134,26 @@ export function SitesDashboard( { queryParams: { search, status } }: SitesDashbo
 	);
 }
 
-function handleTabSelect( tabName: string ) {
-	page(
-		'all' === tabName
-			? removeQueryArgs( window.location.pathname + window.location.search, 'status' )
-			: addQueryArgs( window.location.pathname + window.location.search, { status: tabName } )
-	);
-}
-
-function handleSearch( rawTerm: string ) {
-	const trimmedTerm = rawTerm.trim();
-	if ( trimmedTerm.length ) {
-		page(
-			addQueryArgs( window.location.pathname + window.location.search, { search: trimmedTerm } )
-		);
+/**
+ * Updates a query param used by the sites dashboard, causing a page navigation.
+ * Param will be removed if it is empty or matches its default value.
+ *
+ * @param paramName name of the param being updated
+ * @param paramValue new value for the param
+ * @param defaultValue if the new param matches the default it'll be removed
+ */
+function handleQueryParamChange(
+	paramName: keyof SitesDashboardQueryParams,
+	paramValue: string | null,
+	defaultValue = ''
+) {
+	// Ensure we keep existing query params by appending `.search`
+	const pathWithQuery = window.location.pathname + window.location.search;
+
+	const trimmedValue = paramValue?.trim();
```

Comment:
I think sanitizing the value should be a responsibility of the caller, so we can keep this function as minimal as possible.

Reply:
> I think sanitizing the value should be a responsibility of the caller, so we can keep this function as minimal as possible.

Updated with 250537e2f505e988ae1ae39cdbeb904da1e04b01